spark-pipeline-engine/src/main/scala/com/acxiom/pipeline/fs/SFTPFileManager.scala

```diff
@@ -0,0 +1,193 @@
+package com.acxiom.pipeline.fs
+
+import java.io.{BufferedInputStream, BufferedOutputStream, InputStream, OutputStream}
+
+import scala.collection.JavaConversions.collectionAsScalaIterable
+import com.jcraft.jsch.{ChannelSftp, JSch, SftpException}
+
+object SFTPFileManager {
+  val DEFAULT_PORT = 22
+  val DEFAULT_BULK_REQUESTS = 128
+  val DEFAULT_TRANSFER_BUFFER = 32768
+  val DEFAULT_INPUT_BUFFER = 65536
+  val DEFAULT_OUTPUT_BUFFER = 65536
+}
+
+class SFTPFileManager(user: String,
+                      hostName: String,
+                      port: Int = SFTPFileManager.DEFAULT_PORT,
+                      password: Option[String] = None,
+                      knownHosts: Option[String] = None,
+                      bulkRequests: Int = SFTPFileManager.DEFAULT_BULK_REQUESTS,
+                      config: Option[Map[String, String]] = None
+                     ) extends FileManager {
+  private val jsch = new JSch()
+  if (knownHosts.isDefined) {
+    jsch.setKnownHosts(knownHosts.get)
+  }
+  private lazy val session = {
+    val ses = jsch.getSession(user, hostName, port)
+    if (password.isDefined) {
+      ses.setPassword(password.get)
+    }
+    if (config.isDefined) {
+      ses.setConfig((new java.util.Properties /: config.get) { case (props, (k, v)) => props.put(k, v); props })
+    }
+    ses
+  }
+
+  private lazy val channel: ChannelSftp = {
+    session.connect()
+    val chan = session.openChannel("sftp").asInstanceOf[ChannelSftp]
+    chan.setBulkRequests(bulkRequests)
+    chan
+  }
+
+  /**
+    * Connect to the file system
+    */
+  override def connect(): Unit = {
+    channel.connect()
```

Comment:
Add a check here to see if there is already a connection

Reply:
Done